bubuku/daemon.py

```diff
@@ -51,28 +50,27 @@ def main():
     kafka_properties = KafkaProperties(config.kafka_settings_template,
                                        '{}/config/server.properties'.format(config.kafka_dir))
 
-    amazon = Amazon()
-
-    address_provider = AWSExhibitorAddressProvider(amazon, config.zk_stack_name)
+    env_provider = EnvProvider.create_env_provider(config.development_mode)
+    address_provider = env_provider.get_address_provider(config.zk_stack_name)
 
     _LOG.info("Loading exhibitor configuration")
     buku_proxy = load_exhibitor_proxy(address_provider, config.zk_prefix)
 
     _LOG.info("Loading broker_id policy")
-    broker_id_manager = get_broker_id_policy(config.id_policy, buku_proxy, kafka_properties, amazon)
+    broker_id_manager = get_broker_id_policy(config.id_policy, buku_proxy, kafka_properties, env_provider)
```

Comment:
It makes sense to exclude id_policy from configuration and move `get_broker_id_policy` to `EnvProvider.create_broker_id_manager`, because on AWS wwe are always using 'ip' and for local deployments - 'local'